samza-core/src/main/java/org/apache/samza/application/TaskApplicationDescriptorImpl.java

```diff
@@ -62,7 +62,6 @@ public void setTaskFactory(TaskFactory factory) {
 
   @Override
   public void addInputStream(InputDescriptor inputDescriptor) {
-    // TODO: SAMZA-1841: need to add to the broadcast streams if inputDescriptor is for a broadcast stream
```

Comment:
Same here, we need to add the inputDescriptor to broadcastStreams set if inputDescriptor.isBroadcast() == true